src/components/DragGroup.js

```diff
@@ -0,0 +1,136 @@
+import Piece from "./Piece";
+import React from "react";
+import { dragDestinationOnBoard } from "./Board";
+import { dragDestinationInPool } from "./Pool";
+
+// This component is mounted each time a drag starts.
+export default function DragGroup({ dispatchGameState, gameState }) {
+  const dragState = gameState.dragState;
+  const isShifting = dragState.isShifting;
+  const draggedPieces = gameState.pieces.filter((piece) =>
+    dragState.pieceIDs.includes(piece.id)
+  );
+
+  // Capture the pointer.
+  const dragGroup = React.useRef(null);
+  React.useEffect(() => {
+    let ok = true;
+    try {
+      dragGroup.current.setPointerCapture(dragState.pointerID);
+    } catch (exc) {
+      console.warn("Failed to capture pointer:", exc);
+      ok = false;
+    }
+    ok &&= dragGroup.current.hasPointerCapture(dragState.pointerID);
+    if (!ok) {
+      dispatchGameState({ action: "dragEnd" });
+    }
+    return () => {
+      if (ok && dragGroup.current) {
+        try {
+          dragGroup.current.releasePointerCapture(dragState.pointerID);
+        } catch (exc) {
+          // The pointer is invalid. Normal on touch screens. Ignore it.
+        }
+      }
+    };
+  }, [dragGroup, dragState.pointerID]);
```

Comment:
Hmm, I need to read that. I think it's correct to remove `dragGroup` from this list, but I'm a little surprised it matters.

I'm no expert. Most of what I know about React I learned by reading your code! 🤣

Reply:
OK, I think I understand a little better now. I do take issue with one thing in the Medium post. It says:

> The code looks straightforward at the top, but under the hood, it’s creating a bug just because of `ref.current` in the list of dependencies.

and the code is

```js
useEffect(() => {
  setHeading(`First Heading Value: ${minus}`);
}, [ref.current]);
```

But removing `ref.current` would not fix the bug! The fix is rather to add `minus`.

Still, you're right it's not good to include `dragGroup` here. It's gone now.